src/Http/Http.Extensions/gen/RequestDelegateGenerator.cs

```diff
@@ -39,83 +41,141 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
-                {{StaticRouteHandlerModelEmitter.EmitFilteredRequestHandler()}}
+{{endpoint.EmitRequestHandler()}}
+{{StaticRouteHandlerModelEmitter.EmitFilteredRequestHandler()}}
 
-                RequestDelegate targetDelegate = filteredInvocation is null ? RequestHandler : RequestHandlerFiltered;
-                var metadata = inferredMetadataResult?.EndpointMetadata ?? ReadOnlyCollection<object>.Empty;
-                return new RequestDelegateResult(targetDelegate, metadata);
-            }),
+                    RequestDelegate targetDelegate = filteredInvocation is null ? RequestHandler : RequestHandlerFiltered;
+                    var metadata = inferredMetadataResult?.EndpointMetadata ?? ReadOnlyCollection<object>.Empty;
+                    return new RequestDelegateResult(targetDelegate, metadata);
+                }),
 """);
 
-        var stronglyTypedEndpointDefinitions = endpoints.Select((endpoint, _) => $$"""
-        internal static global::Microsoft.AspNetCore.Builder.RouteHandlerBuilder {{endpoint.HttpMethod}}(
-            this global::Microsoft.AspNetCore.Routing.IEndpointRouteBuilder endpoints,
-            [global::System.Diagnostics.CodeAnalysis.StringSyntax("Route")] string pattern,
-            global::{{StaticRouteHandlerModelEmitter.EmitHandlerDelegateType(endpoint)}} handler,
-            [global::System.Runtime.CompilerServices.CallerFilePath] string filePath = "",
-            [global::System.Runtime.CompilerServices.CallerLineNumber]int lineNumber = 0)
-        {
-            return global::Microsoft.AspNetCore.Http.Generated.GeneratedRouteBuilderExtensionsCore.MapCore(endpoints, pattern, handler, {{StaticRouteHandlerModelEmitter.EmitVerb(endpoint)}}, filePath, lineNumber);
-        }
-""");
+        var stronglyTypedEndpointDefinitions = endpoints
+            .Collect()
+            .Select((endpoints, _) =>
+            {
+                var dedupedByDelegate = endpoints.Distinct(new LambdaComparer<Endpoint>((a, b) =>
+                {
+                    if (a.Response.IsAwaitable == b.Response.IsAwaitable &&
+                        a.Response.IsVoid == b.Response.IsVoid &&
+                        SymbolEqualityComparer.Default.Equals(a.Response.ResponseType, b.Response.ResponseType) &&
+                        a.HttpMethod == b.HttpMethod)
+                    {
+                        return 0;
+                    }
+                    return -1;
+                }, (endpoint) =>
+                {
+                    unchecked
+                    {
+                        var hashCode = SymbolEqualityComparer.Default.GetHashCode(endpoint.Response.ResponseType);
+                        hashCode = (hashCode * 397) ^ endpoint.Response.IsAwaitable.GetHashCode();
+                        hashCode = (hashCode * 397) ^ endpoint.Response.IsVoid.GetHashCode();
+                        hashCode = (hashCode * 397) ^ endpoint.HttpMethod.GetHashCode();
+                        return hashCode;
```

Comment:
We could use the [Microsoft.Bcl.HashCode](https://nuget.org/packages/Microsoft.Bcl.HashCode) package here.

```suggestion
                        return HashCode.Combine(
                            SymbolEqualityComparer.Default.GetHashCode(endpoint.Response.ResponseType),
                            endpoint.Response.IsAwaitable,
                            endpoint.Response.IsVoid,
                            endpoint.HttpMethod);
```